src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,175 @@
+        }
+        head = newNode;
+        sizeList++;
+    }
+
+    public void addToMiddle(T value, Node<T> currentNode) {
+        Node<T> newNode = new Node<>(currentNode.prev, value, currentNode);
+        currentNode.prev.next = newNode;
+        currentNode.prev = newNode;
+        sizeList++;
+    }
+
+    public void addToEnd(T value) {
+        Node<T> lastListNode = tail;
+        Node<T> newNode = new Node<>(tail,value,null);
+        tail = newNode;
+        if (lastListNode == null) {
+            head = newNode;
+        } else {
+            lastListNode.next = newNode;
+        }
+        sizeList++;
+    }
+
     @Override
     public void add(T value) {
+        addToEnd(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkElementPosition(index);
+        int secondIndexList = 1;
+        if (index == sizeList) {
+            addToEnd(value);
+        } else if (index < secondIndexList) {
+            addAtTheFront(value);
+        } else {
+            addToMiddle(value,getCurrentNode(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkElementExistence(index);
+        Node<T> lastListNode = tail;
+        for (int i = sizeList - 1; i > index; i--) {
+            lastListNode = lastListNode.prev;
+        }
+        return lastListNode.element;
+    }
+
+    public Node<T> getCurrentNode(int index) {
+        checkElementExistence(index);
+        Node<T> lastNode = tail;
+        int i = sizeList - 1;
+        while (i > index) {
+            lastNode = lastNode.prev;
+            i--;
+        }
+        return lastNode;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkElementExistence(index);
+        Node<T> nodeToBeReplaced = getCurrentNode(index);
+        T replacedNodeReturn = nodeToBeReplaced.element;
+        nodeToBeReplaced.element = value;
+        return replacedNodeReturn;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkElementExistence(index);
+        return unlink(getCurrentNode(index));
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node<T> toBeRemoved = head;
+        int index = 0;
+        while (toBeRemoved.next != null && toBeRemoved.element != object) {
+            toBeRemoved = toBeRemoved.next;
+            index++;
+            if ((toBeRemoved.element == object)
+                    || (toBeRemoved.element
+                    != null && toBeRemoved.element.equals(object))) {
+                remove(index);
+                return true;
+            }
+        }
+        if (toBeRemoved.element == object) {
+            remove(index);
+            return true;
+        } else {
+            return false;
+        }
```

Comment:
```suggestion
        Node<T> toBeRemoved = head;
        while (toBeRemoved != null) {
            if (Objects.equals(toBeRemoved.element, object)) {
                unlink(toBeRemoved);
                return true;
            }
            toBeRemoved = toBeRemoved.next;
        }
        return false;
```

we can make it simpler

Reply:
@MaliukDaria Большое спасибо за подсказки и исправления.